### PR TITLE
docs(components): accordion controlled example shows errors

### LINKED ISSUE #2337: [BUG] - Accordion Controlled Example shows errors
### NextUI Version

2.2.9

### Describe the bug

When following the [controlled accordion example](https://nextui.org/docs/components/accordion#controlled) I am seeing the following error despite the functionality working as expected.
```
Type 'Dispatch<SetStateAction<Set<string>>>' is not assignable to type '(keys: Selection) => any'.
  Types of parameters 'value' and 'keys' are incompatible.
    Type 'Selection' is not assignable to type 'SetStateAction<Set<string>>'.
      Type 'string' is not assignable to type 'SetStateAction<Set<string>>'.ts(2322)
```

Changing the erroring line to the following resolves the issue as a workaround:
```
onSelectionChange={(keys: Selection) => setSelectedAccordionKeys(keys as Set<string>)}
```

### Your Example Website or App

_No response_

### Steps to Reproduce the Bug or Issue

1. Follow controlled example
2. View error

### Expected behavior

Controlled example works without errors

### Screenshots or Videos

image

### Operating System Version

macOS

### Browser

Chrome

### PR DESCRIPTION
**Pull request by kuri-sun**
Closes #2337

## 📝 Description

After you copy and paste the code from [controlled accordion example](https://nextui.org/docs/components/accordion#controlled), you will an error on the `onSelectionChange` function.
old
Because we are missing the type for the `useState`. 

## ⛳️ Current behavior (updates)

Throwing the following error
```
Type 'Dispatch<SetStateAction<Set<string>>>' is not assignable to type '(keys: Selection) => any'.
  Types of parameters 'value' and 'keys' are incompatible.
    Type 'Selection' is not assignable to type 'SetStateAction<Set<string>>'.
      Type 'string' is not assignable to type 'SetStateAction<Set<string>>'.ts(2322)
```

## 🚀 New behavior

No error.
new

## 💣 Is this a breaking change (Yes/No):

No.

### CODE REVIEW

File: apps/docs/content/components/accordion/controlled.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 const App = `import {Accordion, AccordionItem} from "@nextui-org/react";
```
Comment (jrgarciadev) [inline apps/docs/content/components/accordion/controlled.ts:1]: @kuri-sun please add the `Selection` import and create a reactTs for the typescript code.

Example:

```ts


const AppTs = `import type {Selection} from "@nextui-org/react";

import {Accordion, AccordionItem} from "@nextui-org/react";

export default function App() {
  const [selectedKeys, setSelectedKeys] = React.useState<Selection>(new Set(["1"]));
  const defaultContent =
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.";
  return (
    <Accordion
      selectedKeys={selectedKeys}
      onSelectionChange={setSelectedKeys}
    >
      <AccordionItem key="1" aria-label="Accordion 1" title="Accordion 1">
        {defaultContent}
      </AccordionItem>
      <AccordionItem key="2" aria-label="Accordion 2" title="Accordion 2">
        {defaultContent}
      </AccordionItem>
      <AccordionItem key="3" aria-label="Accordion 3" title="Accordion 3">
        {defaultContent}
      </AccordionItem>
    </Accordion>
  );
}`;

const react = {
  "/App.jsx": App,
};

const reactTs = {
  "/App.tsx": AppTs,
};

export default {
  ...react,
  ...reactTs,
};
```

**Review by jrgarciadev — CHANGES_REQUESTED** [review]

File: apps/docs/content/components/accordion/controlled.ts (hunk 1 above)
Comment (kuri-sun) [reply apps/docs/content/components/accordion/controlled.ts:1]: @jrgarciadev Thank you! I will do this right away!

**Comment by kuri-sun** [comment]
@jrgarciadev Thanks for the review! The change is applied! 👍

File: apps/docs/content/components/accordion/controlled.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
 
 export default function App() {
-  const [selectedKeys, setSelectedKeys] = React.useState(new Set(["1"]));
+  const [selectedKeys, setSelectedKeys] = React.useState<Selection>(new Set(["1"]));
```
Comment (jrgarciadev) [inline apps/docs/content/components/accordion/controlled.ts:4]:
```suggestion
  const [selectedKeys, setSelectedKeys] = React.useState(new Set(["1"]));
```

As this file is meant to be used as plain "Javascript" file we cannot add typescript code

**Review by jrgarciadev — CHANGES_REQUESTED** [review]

**Review by jrgarciadev — APPROVED** [review]

**Pull request merged**